src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCITriggerService.java

```diff
@@ -58,7 +58,7 @@ public void triggerBuild(ProgrammingExerciseParticipation participation, String
             // The 'user' is not properly logged into Artemis, this leads to an issue when accessing custom repository methods.
             // Therefore, a mock auth object has to be created.
             SecurityUtils.setAuthorizationObject();
-            Result result = programmingExerciseGradingService.processNewProgrammingExerciseResult(participation, buildResult).orElseThrow();
+            Result result = programmingExerciseGradingService.processNewProgrammingExerciseResult(participation, buildResult);
             programmingMessagingService.notifyUserAboutNewResult(result, participation);
```

Comment:
Is the `null`/error case covered? Previously we threw an exception in this case (`.orElseThrow()`), now we just pass it to the notification function.

Quickly looking at the implementation this might lead to a nullpointer exception further down in case of an error and hence null value.